board/boards/red.h

```diff
@@ -0,0 +1,241 @@
+      break;
+    default:
+      break;
+  }
+}
+
+void red_enable_can_transceivers(bool enabled) {
+  for(uint8_t i=1U; i<=4U; i++){
+    // Leave main CAN always on for CAN-based ignition detection
+    if((car_harness_status == HARNESS_STATUS_FLIPPED) ? (i == 3U) : (i == 1U)){
+      red_enable_can_transceiver(i, true);
+    } else {
+      red_enable_can_transceiver(i, enabled);
+    }
+  }
+}
+
+void red_set_led(uint8_t color, bool enabled) {
+  switch (color){
+    case LED_RED:
+      set_gpio_output(GPIOE, 4, !enabled);
+      break;
+     case LED_GREEN:
+      set_gpio_output(GPIOE, 3, !enabled);
+      break;
+    case LED_BLUE:
+      set_gpio_output(GPIOE, 2, !enabled);
+      break;
+    default:
+      break;
+  }
+}
+
+void red_set_gps_load_switch(bool enabled) {
+  // No GPS on red panda
+  UNUSED(enabled);
+}
+
+void red_set_usb_load_switch(bool enabled) {
+  set_gpio_output(GPIOB, 14, !enabled);
+}
+
+void red_set_usb_power_mode(uint8_t mode) {
+  bool valid = false;
+  switch (mode) {
+    case USB_POWER_CLIENT:
+      red_set_usb_load_switch(false);
+      valid = true;
+      break;
+    case USB_POWER_CDP:
+      red_set_usb_load_switch(true);
+      valid = true;
+      break;
+    default:
+      break;
+  }
+  if (valid) {
+    usb_power_mode = mode;
+  }
+}
+
+void red_set_gps_mode(uint8_t mode) {
+  // No GPS on red panda
+  UNUSED(mode);
+}
+
+void red_set_can_mode(uint8_t mode){
+  switch (mode) {
+    case CAN_MODE_NORMAL:
+    case CAN_MODE_OBD_CAN2:
+      if ((bool)(mode == CAN_MODE_NORMAL) != (bool)(car_harness_status == HARNESS_STATUS_FLIPPED)) {
+        // B12,B13: disable normal mode
+        set_gpio_pullup(GPIOB, 12, PULL_NONE);
+        set_gpio_mode(GPIOB, 12, MODE_ANALOG);
+
+        set_gpio_pullup(GPIOB, 13, PULL_NONE);
+        set_gpio_mode(GPIOB, 13, MODE_ANALOG);
+        
+        // B5,B6: FDCAN2 mode
+        set_gpio_pullup(GPIOB, 5, PULL_NONE);
+        set_gpio_alternate(GPIOB, 5, GPIO_AF9_FDCAN2);
+
+        set_gpio_pullup(GPIOB, 6, PULL_NONE);
+        set_gpio_alternate(GPIOB, 6, GPIO_AF9_FDCAN2);
+      } else {
+        // B5,B6: disable normal mode
+        set_gpio_pullup(GPIOB, 5, PULL_NONE);
+        set_gpio_mode(GPIOB, 5, MODE_ANALOG);
+
+        set_gpio_pullup(GPIOB, 6, PULL_NONE);
+        set_gpio_mode(GPIOB, 6, MODE_ANALOG);
+        // B12,B13: FDCAN2 mode
+        set_gpio_pullup(GPIOB, 12, PULL_NONE);
+        set_gpio_alternate(GPIOB, 12, GPIO_AF9_FDCAN2);
+
+        set_gpio_pullup(GPIOB, 13, PULL_NONE);
+        set_gpio_alternate(GPIOB, 13, GPIO_AF9_FDCAN2);
+      }
+      break;
+    default:
+      break;
+  }
+}
+
+void red_usb_power_mode_tick(uint32_t uptime){
+  UNUSED(uptime);
+  // Not applicable
+}
+
+bool red_check_ignition(void){
+  // ignition is checked through harness
+  return harness_check_ignition();
+}
+
+uint32_t red_read_current(void){
+  // No current sense on red panda
+  return 0U;
+}
+
+void red_set_ir_power(uint8_t percentage){
```

Comment:
put these somewhere common. after merging this, we should have the other boards use the common ones too